staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -67,6 +67,22 @@ const timeFmt = "2006-01-02T15:04:05.999"
 // undesired becomes completely unused, all the config objects are
 // read and processed as a whole.
 
+const (
+	// Borrowing among priority levels will be accomplished by periodically
+	// adjusting the current concurrency limits (CurrentCLs);
+	// borrowingAdjustmentPeriod is that period.
+	borrowingAdjustmentPeriod = 10 * time.Second //nolint:golint,unused
+
+	// The input to the borrowing is smoothed seat demand figures.
+	// Every adjustment period, each priority level's new smoothed demand is set to
+	// seatDemandSmoothingCoefficient times the previous smoothed demand
+	// plus (1-seatDemandSmoothingCoefficient) times the recent envelope
+	// of seat demand.  The particular number appearing here has the property
+	// that half of the current smoothed value comes from the smoothed value
+	// of 5 minutes ago.
```

Comment:
I'm a little concerned about start-up, it seems like it takes ~10 minutes to mitigate the impact of the very first sample, which is essentially random.

Steady-state, this seems good and a low-pass filter seems much superior to (for example) a running average for this.

Reply:
Can you please elaborate on the concern?  Remembering that the logic here is asymmetric (quick jump up due to the `max`, slow drift down while the pressure is off).

Reply:
It sounds like this may already be solved by the envelope computation, then.